### PR TITLE
Improve workflow

### PR DESCRIPTION
**Pull request by Prefection**
- Separated Posfix installation
- Dynamic gitlab.rb configuration

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,8 +1,23 @@
 ---
+gitlab_use_external_smtp: false
+gitlab_external_url: "https://gitlab.example.com"
```
Comment (winggundamth) [inline defaults/main.yml]: I don't see usage of this variable

File: defaults/main.yml [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+gitlab_roles: ""
```
Comment (winggundamth) [inline defaults/main.yml]: What is this for? I don't see usage of this variable

File: defaults/main.yml [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+  #   config_value: 443
+  #   config_quote: yes
+  # - config_prefix: unicorn
+  #   config_key: worker_processes
+  #   config_value: 8
+
 gitlab_search_config_path: "{{ playbook_dir }}/files/groups/{{ item }}/gitlab"
-gitlab_search_config_paths: []
 
-gitlab_host_config_file_path: "{{ gitlab_host_config_path | default(inventory_hostname) }}/gitlab.rb.j2"
```
Comment (winggundamth) [inline defaults/main.yml]: For line 3 and 5 can you preserve this for backward compatibility. I have use this a lot.

File: tasks/gitlab_post.yml [hunk 4]
```diff
@@ -1,12 +1,19 @@
 ---
 - name: Copy GitLab configuration file
   template:
-    src: "{{ gitlab_host_config_file_path }}"
+    src: gitlab.rb.j2
```
Comment (winggundamth) [inline tasks/gitlab_post.yml]: Can you preserve this? I need it for backward compatibility for many installation.

File: tasks/gitlab_pre.yml [hunk 5]
```diff
@@ -2,14 +2,20 @@
...
       - Start sshd
+
+  - name: Install Postfix
+    yum:
+      name: postfix
+      state: present
+      update_cache: yes
+    notify:
       - Start postfix
+    when: not gitlab_use_external_smtp
```
Comment (winggundamth) [inline tasks/gitlab_pre.yml]: Should put this outside the block since it's the same as Debian

File: tasks/gitlab_pre.yml [hunk 6]
```diff
@@ -19,21 +25,29 @@
 
   - name: Allow some firewall to using
     firewalld:
-      service: "{{ item.1 }}"
+      service: http
```
Comment (winggundamth) [inline tasks/gitlab_pre.yml]: Do we need https as well?

Comment (opstacloud) [reply tasks/gitlab_pre.yml]: added e4e2d1d

File: defaults/main.yml (hunk 1 above)
Comment (Prefection) [reply defaults/main.yml]: being used by templates/gitlab.rb.j2

File: defaults/main.yml (hunk 2 above)
Comment (Prefection) [reply defaults/main.yml]: being used by templates/gitlab.rb.j2

File: defaults/main.yml (hunk 3 above)
Comment (Prefection) [reply defaults/main.yml]: fixed on 30e15be

File: tasks/gitlab_post.yml (hunk 4 above)
Comment (Prefection) [reply tasks/gitlab_post.yml]: fixed on 30e15be

File: tasks/gitlab_pre.yml (hunk 5 above)
Comment (Prefection) [reply tasks/gitlab_pre.yml]: due to difference Ansible module for RHEL (yum) and Debian (apt) then I need to separated them. (same as existing task)

File: tasks/gitlab_pre.yml (hunk 6 above)
Comment (Prefection) [reply tasks/gitlab_pre.yml]: fixed on e4e2d1d

**Pull request merged**